sale_elaboration/models/sale_order.py

```diff
@@ -70,3 +70,7 @@ def _prepare_invoice_line(self, qty):
         if self.is_elaboration:
             vals['name'] = '{} - {}'.format(self.order_id.name, self.name)
         return vals
+
+    @api.onchange('product_id')
+    def onchange_sale_elaboration_product_id(self):
```

Comment:
Please override standard onchange instead of creating a new one, calling super. Any way, why not putting `is_elaboration` as a related field?

Reply:
Because this line always will be an elaboration although you change it in product for all the lines that were created.
In this case is not needed call to super product_id change